File: Data Structures & Algorithms/two-integer-sum/submission-4.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] toret = new int[2];

        HashMap<Integer, Integer> values = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            values.put(nums[i], i);
        }

        for (int i = 0; i < nums.length; i++) {
            if (values.containsKey(target - nums[i]) && i != values.get(target - nums[i])) {
                toret[0] = Math.min(i, values.get(target-nums[i]));
                toret[1] = Math.max(i, values.get(target-nums[i]));
                return toret;
            }
        }

        return toret;
            
        
    }
}

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-4.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        memo = [1] * len(nums)   

        for i in range(len(nums)):
            j = i - 1
            best = 0
            while j >= 0:
                if nums[j] < nums[i]:
                    best = max(best, memo[j])
                j -= 1
            memo[i] = best + 1
        
        return max(memo)
        

            








        